### ISSUE TITLE
Runtime error when trying to run apps without directory/Image.

### ISSUE THREAD

**Issue by sangee2004** [acorn-io/runtime#1942]
**acorn version** - `v0.8.0-alpha7-98-gbb130866+bb130866`

**Steps to reproduce the problem:**

Deploy app without directory/Image.

```
acorn run -n mytestuu 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x340ce26]

goroutine 1 [running]:
github.com/acorn-io/runtime/pkg/buildclient.Stream({0x484b258?, 0xc000248f00}, {0x47ff510, 0x1}, 0xc001655220, 0xc00120ef98, 0x0?, 0xc0009bdb80)
	github.com/acorn-io/runtime/pkg/buildclient/client.go:38 +0x1a6
github.com/acorn-io/runtime/pkg/client.(*DefaultClient).AcornImageBuild(0xc00023fc70, {0x484b258, 0xc000248f00}, {0xc000234270, 0x9}, 0xbf58a28?)
	github.com/acorn-io/runtime/pkg/client/build.go:112 +0x5ed
github.com/acorn-io/runtime/pkg/client.(*DeferredClient).AcornImageBuild(0xc000a74340, {0x484b258, 0xc000248f00}, {0xc000234270, 0x9}, 0x4069f40?)
	github.com/acorn-io/runtime/pkg/client/deferred.go:339 +0x7a
github.com/acorn-io/runtime/pkg/client.(*MultiClient).AcornImageBuild(0xc000e60570, {0x484b258, 0xc000248f00}, {0xc000234270, 0x9}, 0x0?)
	github.com/acorn-io/runtime/pkg/client/multi.go:453 +0xa6
github.com/acorn-io/runtime/pkg/imagesource.ImageSource.GetImageAndDeployArgs({{0x47ff510, 0x1}, {0xc000234270, 0x9}, {0x0, 0x0, 0x0}, {0x64711c0, 0x0, 0x0}, ...}, ...)
	github.com/acorn-io/runtime/pkg/imagesource/helper.go:190 +0x2a3
github.com/acorn-io/runtime/pkg/cli.(*Run).Run(0xc000924700, 0xc000e2cc00, {0xc0008f41a0, 0x0, 0x2})
	github.com/acorn-io/runtime/pkg/cli/run.go:285 +0x5d2
github.com/acorn-io/runtime/pkg/cli/builder.bind.func1(0xc000e2cc00?, {0xc0008f41a0, 0x0, 0x2})
	github.com/acorn-io/runtime/pkg/cli/builder/builder.go:431 +0x214
github.com/spf13/cobra.(*Command).execute(0xc000e2cc00, {0xc0008f4180, 0x2, 0x2})
	github.com/spf13/cobra@v1.7.0/command.go:940 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0xc000db8600)
	github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.7.0/command.go:992
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	github.com/spf13/cobra@v1.7.0/command.go:985
github.com/acorn-io/runtime/pkg/cli.RunAndHandleError({0x484b258?, 0xc000248f00?}, 0xc000db8600?)
	github.com/acorn-io/runtime/pkg/cli/errors.go:17 +0x5d
main.main()
	github.com/acorn-io/runtime/main.go:15 +0x2f
```

**Comment by cjellick** [comment]
seems like this might be fixed, but i'll let you take a quick look @keyallis

**Comment by g-linville** [comment]
I believe that when no image or directory or Acornfile is specified, the current directory `.` is assumed.

**Comment by keyallis** [comment]
The issue didn't require any changes to fix the observed behavior, but I added an additional safeguard to try and prevent it from hitting panic again so that (hopefully) it will fail more gracefully if it happens again.
I also went ahead and added the `acorn run` command without directory/image to the unit test cases.

TLDR: Not much to QA here

**Comment by sangee2004** [comment]
Not able to reproduce this issue when testing with recent builds .Tested with acorn version v0.8.0-alpha7-151-gc1b7f440+c1b7f440.

Able to deploy apps successfully using `acorn run -n <app-name>`
